dev/io.openliberty.microprofile.openapi.2.0.internal.servlet/src/io/openliberty/microprofile/openapi20/internal/servlet/ApplicationServlet.java

```diff
@@ -58,13 +61,16 @@ public void init(ServletConfig config) throws ServletException {
         appRegistryTracker.open();
         modelOperationsTracker = new ServiceTracker<>(bundleContext, OpenAPIModelOperations.class, null);
         modelOperationsTracker.open();
+        versionConfigTracker = new ServiceTracker<>(bundleContext, OpenAPIVersionConfig.class, null);
```

Comment:
should the init and destroy code for OpenAPIVersionConfig be beta guarded as well?

Reply:
Yes, we do beta-guard the init code in the [activate method of OpenAPIVersionConfigImpl](https://github.com/OpenLiberty/open-liberty/pull/29877/files#diff-aa40fd44bd64ea333e2485f56da55fb2588a9cd290035c6d3519bce144ab5804R50-R56). There is no destroy code for that class.

Here we're just setting up a tracker for that service. It doesn't actually create a dependency on the service, it just keeps track of whether it's present. We do beta-guard around attempting to retrieve and call the service further down this file.